src/hipscat/inspection/almanac_catalog_info.py

```diff
@@ -0,0 +1,149 @@
+import dataclasses
+import os
+from dataclasses import dataclass, field
+from typing import List
+
+import yaml
+from typing_extensions import Self
+
+from hipscat.catalog.dataset.base_catalog_info import BaseCatalogInfo
+from hipscat.catalog.dataset.catalog_info_factory import (create_catalog_info,
+                                                          from_catalog_dir)
+from hipscat.io import file_io
+
+
+@dataclass
+class AlmanacCatalogInfo:
+    """Container for parsed almanac information."""
+
+    file_path: str = ""
+    namespace: str = ""
+    catalog_path: str = ""
+    catalog_name: str = ""
+    catalog_type: str = ""
+    primary: str = ""
+    join: str = ""
+    primary_link: Self = None
+    join_link: Self = None
+    sources: List[Self] = field(default_factory=list)
+    objects: List[Self] = field(default_factory=list)
+    margins: List[Self] = field(default_factory=list)
+    associations: List[Self] = field(default_factory=list)
+    associations_right: List[Self] = field(default_factory=list)
+    indexes: List[Self] = field(default_factory=list)
+
+    creators: List[str] = field(default_factory=list)
+    description: str = ""
+    version: str = ""
+    deprecated: str = ""
+
+    catalog_info: dict = field(default_factory=dict)
+
+    catalog_info_object: BaseCatalogInfo = None
+
+    def __post_init__(
+        self,
+    ):
+        if len(self.catalog_info):
+            self.catalog_info_object = create_catalog_info(self.catalog_info)
+
+        ## Allows use of $HIPSCAT_DEFAULT_DIR in paths
+        self.catalog_path = os.path.expandvars(self.catalog_path)
+
+    def get_primary_text(self) -> str:
+        """Find some text name/path for a primary catalog link, if present"""
+        if self.primary:
+            return self.primary
+        if self.catalog_info and "primary_catalog" in self.catalog_info:
+            return self.catalog_info["primary_catalog"]
```

Comment:
Could this be in the init? So that if we have a `catalog_info` and no `primary` then `primary` can be set? Unless you expect the object to be modified?